tests/php/src/Admin/PairedBrowsingTest.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * Tests for PairedBrowsing class.
+ *
+ * @package AmpProject\AmpWP\Tests
+ */
+
+namespace AmpProject\AmpWP\Tests\Admin;
+
+use AMP_Options_Manager;
+use AMP_Theme_Support;
+use AmpProject\AmpWP\Admin\PairedBrowsing;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Option;
+use AmpProject\AmpWP\QueryVar;
+use AmpProject\AmpWP\Tests\DependencyInjectedTestCase;
+use AmpProject\AmpWP\Tests\Helpers\AssertContainsCompatibility;
+use AmpProject\DevMode;
+use WPDieException;
+use WP_Admin_Bar;
+
+/** @coversDefaultClass \AmpProject\AmpWP\Admin\PairedBrowsing */
+class PairedBrowsingTest extends DependencyInjectedTestCase {
+
+	use AssertContainsCompatibility;
+
+	/** @var PairedBrowsing */
+	private $instance;
+
+	public function setUp() {
+		parent::setUp();
+		$this->instance = $this->injector->make( PairedBrowsing::class );
+	}
+
+	/** @covers ::is_needed() */
+	public function test_is_needed() {
+		AMP_Options_Manager::update_option( Option::THEME_SUPPORT, AMP_Theme_Support::STANDARD_MODE_SLUG );
+		$this->assertFalse( PairedBrowsing::is_needed() );
+
+		AMP_Options_Manager::update_option( Option::THEME_SUPPORT, AMP_Theme_Support::TRANSITIONAL_MODE_SLUG );
```

Comment:
Maybe add an assertion for `AMP_Theme_Support::READER_MODE_SLUG` just for completeness: 

```suggestion
		AMP_Options_Manager::update_option( Option::THEME_SUPPORT, AMP_Theme_Support::READER_MODE_SLUG );
		$this->assertFalse( PairedBrowsing::is_needed() );

				AMP_Options_Manager::update_option( Option::THEME_SUPPORT, AMP_Theme_Support::TRANSITIONAL_MODE_SLUG );
```